### PR TITLE
CLI: Add shell auto completions

### PR DESCRIPTION
**Pull request by paranoidaditya**
Adds a subcommand to dump out bash and zsh autocompletion files

### CODE REVIEW

File: cmd/kind/completion/bash/bash.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2018 The Kubernetes Authors.
```
Comment (tao12345666333) [inline cmd/kind/completion/bash/bash.go]: change it to 2019. [1]

File: cmd/kind/completion/completion.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2018 The Kubernetes Authors.
```
Comment (tao12345666333) [inline cmd/kind/completion/completion.go]: [1]

File: cmd/kind/completion/zsh/zsh.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2018 The Kubernetes Authors.
```
Comment (tao12345666333) [inline cmd/kind/completion/zsh/zsh.go]: [1]

**Comment by paranoidaditya** [comment]
Autocompletion is definitely useful imo, we could either do it this way or have static files up for grabs?

**Comment by BenTheElder** [comment]
yep, apologies for the latency, couple other things have been eating into review time etc.

reading it now the code looks great, thanks.

**Comment by BenTheElder** [comment]
@neolit123 maybe we should just add this and later we can improve the root command help to organize command groupings?

**Comment by BenTheElder** [comment]
/hold cancel
thanks again @paranoidaditya !

**Pull request merged**